### PR TITLE
SSB4: Bugfixes and balance changes

### PR DESCRIPTION
**Pull request by xInstruct**

### CODE REVIEW

File: data/mods/ssb/moves.ts [hunk 1]
```diff
@@ -3502,69 +3447,49 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+			this.add('-anim', source, "Trick", target);
+		},
+		onHit(target, source, effect) {
+			const item = target.takeItem();
+			if (!target.item) {
+				if (item) this.add('-enditem', target, item.name, '[from] move: Trickery', '[of] ' + source);
+				const items: ModdedItemData[] = [];
+				for (const id in Items) {
+					const itemName = Items[id];
+					items.push(itemName);
```
Comment (MacChaeger) [inline data/mods/ssb/moves.ts:3475]: Should this check if the item `isNonstandard`? Berserk Gene or Crucibellite might be a bit weird to get.

File: data/mods/ssb/abilities.ts [hunk 2]
```diff
@@ -1621,16 +1634,15 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
...
-			for (const move of moves) {
-				types.push(this.dex.getMove(move).type);
-			}
-			let type = this.sample(types);
-			while (!pokemon.setType(type)) {
+			const types = pokemon.moveSlots.map(slot => this.dex.getMove(slot.id).type);
+			let type = '???';
+			if (types.length) {
 				type = this.sample(types);
 			}
```
Comment (KrisXV) [inline data/mods/ssb/abilities.ts:1641]:
```suggestion
			const type = types.length ? this.sample(types) : '???';
```

File: data/mods/ssb/moves.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import {getName} from './conditions';
+import {Items} from './items';
```
Comment (KrisXV) [inline data/mods/ssb/moves.ts:2]: no

File: data/mods/ssb/moves.ts [hunk 4]
```diff
@@ -3502,69 +3447,49 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+		},
+		onHit(target, source, effect) {
+			const item = target.takeItem();
+			if (!target.item) {
+				if (item) this.add('-enditem', target, item.name, '[from] move: Trickery', '[of] ' + source);
+				const items: ModdedItemData[] = [];
+				for (const id in Items) {
+					const itemName = Items[id];
+					items.push(itemName);
 				}
```
Comment (KrisXV) [inline data/mods/ssb/moves.ts:3476]:
```suggestion
				const items = Object.keys(this.dex.data.Items).map(obj => this.dex.getItem(obj).name);
```

File: data/mods/ssb/moves.ts [hunk 5]
```diff
@@ -3502,69 +3447,49 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
-						this.add('-message', `${pokemon.name} was hurt by the shifting rocks!`);
-					}
-					this.effectData.damage = 7;
-					pokemon.side.removeSideCondition(`shiftingrocks`);
-					return false;
+				let randomItem = '';
+				if (items.length) {
+					// @ts-ignore
+					randomItem = this.sample(items).name;
 				}
```
Comment (KrisXV) [inline data/mods/ssb/moves.ts:3481]:
```suggestion
				if (items.length) randomItem = this.sample(items);
```

**Review by KrisXV — CHANGES_REQUESTED** [review]

File: data/mods/ssb/moves.ts (hunk 3 above)
Comment (xInstruct) [reply data/mods/ssb/moves.ts:2]: oop lol

File: data/mods/ssb/moves.ts (hunk 1 above)
Comment (xInstruct) [reply data/mods/ssb/moves.ts:3477]: atm I assume we just go full meme with randomizations regardless.

Comment (KrisXV) [reply data/mods/ssb/moves.ts:3477]: I don't think perish song specified any bounds for his custom move, anything from mail to crucibellite is currently fine

Comment (xInstruct) [reply data/mods/ssb/moves.ts:3477]: will talk with perish song, theyre the final verdict on this.

**Review by KrisXV — APPROVED** [review]

File: data/mods/ssb/moves.ts [hunk 6]
```diff
@@ -2700,8 +2646,8 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
-		shortDesc: "Transform into Unown. Unown: Transform to mon.",
+		desc: "The user changes into a random Pokemon with a first name letter that matches the forme Unown is currently in (A -> Alakazam, etc) that has base stats that would benefit from Unown's EV/IV/Nature spread and moves. Using it while in a forme that is not Unown will make it revert back to the Unown forme it transformed in (If an Unown transforms into Alakazam, it'll transform back to Unown-A when used again). Light of Ruin becomes Strange Steam, Psystrike becomes Psyshock, Secret Sword becomes Aura Sphere, Mind Blown becomes Flamethrower, and Seed Flare becomes Apple Acid while in a non-Unown forme. This move's type varies based on the user's primary type.",
+		shortDesc: "Transform into Unown/mon. Type = users type.",
```
Comment (KrisXV) [inline data/mods/ssb/moves.ts:2650]:
```suggestion
		shortDesc: "Transform into Unown/mon. Type=user 1st type.",
```

**Pull request merged**